apps-script/3p-resources/3p-resources.gs

```diff
@@ -0,0 +1,259 @@
+/**
+ * Copyright 2024 Google LLC
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+// [START add_ons_preview_link]
+// [START add_ons_case_preview_link]
+
+/**
+* Entry point for a support case link preview.
+*
+* @param {!Object} event The event object.
+* @return {!Card} The resulting preview link card.
+*/
+function caseLinkPreview(event) {
+
+  // If the event object URL matches a specified pattern for support case links.
+  if (event.docs.matchedUrl.url) {
+
+    // Uses the event object to parse the URL and identify the case details.
+    const caseDetails = parseQuery(event.docs.matchedUrl.url);
+
+    // Builds a preview card with the case name, and description
+    const caseHeader = CardService.newCardHeader()
+      .setTitle(`Case ${caseDetails["name"][0]}`);
+    const caseDescription = CardService.newTextParagraph()
+      .setText(caseDetails["description"][0]);
+
+    // Returns the card.
+    // Uses the text from the card's header for the title of the smart chip.
+    return CardService.newCardBuilder()
+      .setHeader(caseHeader)
+      .addSection(CardService.newCardSection().addWidget(caseDescription))
+      .build();
+  }
+}
+
+/**
+* Extracts the URL parameters from the given URL.
+*
+* @param {!string} url The URL to parse.
+* @return {!Map} A map with the extracted URL parameters.
+*/
+function parseQuery(url) {
```

Comment:
Can we use something like https://developer.mozilla.org/en-US/docs/Web/API/URL to parse the URL?

Reply:
`URL` is not available from Apps Script so I need to define custom utility functions instead.